packages/pluggableWidgets/rich-text-web/src/components/RichText.tsx

```diff
@@ -56,51 +68,70 @@ export const RichTextEditor = (props: RichTextProps): ReactElement => {
         },
         initContent: value,
         dispatchEvent: ({ type, payload }) => {
+            if (type === CKEditorEventAction.key) {
+                if (props.onKeyPress) {
+                    props.onKeyPress();
+                }
+            }
             if (type === CKEditorEventAction.change) {
                 const value = payload.editor.getData();
                 if (props.onKeyChange) {
                     props.onKeyChange();
                 }
-                if (props.onKeyPress) {
-                    props.onKeyPress();
-                }
                 if (props?.onValueChange) {
                     const content = props.sanitizeContent ? sanitizeHtml(value) : value;
+                    localEditorValueRef.current = content;
                     props?.onValueChange(content);
                 }
             }
         },
-        subscribeTo: ["change"]
+        subscribeTo: ["change", "key"]
     });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
If i am not mistaken this rule is a warn in eslint, which doesnt need to be disabled.

Reply:
@diego-antonelli, can you explain bit more in deep as I can't get what needs to be disabled